### PR TITLE
factor hook offset into stem slash calculations

### PR DESCRIPTION
**Pull request by miiizen**
Resolves: #19481 

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Comment by zacjansheski** [comment]
Tested on Mac OS 11.6.5, Ubuntu 20.04.6, Windows 11
Approved (pending @mike-spa 's approval)

File: src/engraving/rendering/dev/tlayout.cpp
Comment (cbjeukendrup) [inline src/engraving/rendering/dev/tlayout.cpp:4824]: Maybe it would be better to replace `hook->pos()` with `hook->layoutData()->pos()`.

Comment (miiizen) [reply src/engraving/rendering/dev/tlayout.cpp:4824]: Ah yes - I hadn't noticed that `hook->pos()` included the offset, thanks!

**Review by cbjeukendrup — APPROVED** [review]

**Comment by zacjansheski** [comment]
Is this waiting approval from @mike-spa or is it ready to merge?

**Review by mike-spa — APPROVED** [review]
Looks good to me!

**Comment by cbjeukendrup** [comment]
(rebased, because of layoutData/ldata conflict)

**Pull request merged**
